### PR TITLE
Assign font family to input fields

### LINKED ISSUE #2094: Some form inputs don't use font settings
**Describe the current behavior**
Some form elements don't have any font settings applied, so they default to the Arial font.

**Describe the expected behavior**
All form elements should have the font settings applied

**Additional context/screenshots**
https://screenshot.click/02-42-mkh9b-3fb6m.png
https://screenshot.click/02-42-63atc-91j2k.png

### LINKED ISSUE #2702: Input tags font
## Describe the current behavior

All input tags are using browser's web safe font

## Describe the expected behavior

Input tags should use the theme's body font. 

## Version information (Dawn, browsers and operating systems)

- Dawn Version: 10.0.0
- Chrome Version 114.0.5735.110
- macOS Version 16

## Possible solution
.field__input {
    font-family: var(--font-body-family);
    font-style: var(--font-body-style);
    font-weight: var(--font-body-weight);
}

### PR DESCRIPTION
**Pull request by kjellr**
### PR Summary: 

- Ensures that input fields use the body font family. 

### Why are these changes introduced?

Fixes #2702
Fixes #2094

Input fields don't natively inherit the font family from `body`, and we don't assign a font family to most of our input fields. This results in them falling back to Arial, which looks really inconsistent. 

It's especially noticeable in the contact form, where the "Comment" field has the body font family assignment, and the rest fall back to Arial. 

We should clean this up. 

### What approach did you take?

I just added a font family assignment to our usual input field rules. We could alternatively assign the font family variable directly to the element tags (e.g. `input, select, textarea, button { var(--font-body-family); }`), or even just set `font-family: inherit` there. But this worked for now and was the least aggressive approach. 

### Visual impact on existing themes

Merchants' body font family will be used for form inputs. 

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/140310577174/editor)

Before|After
---|---
Screenshot 2023-07-19 at 8 14 49 AM|Screenshot 2023-07-19 at 8 12 48 AM

Before|After
---|---
Screenshot 2023-07-19 at 8 15 20 AM|Screenshot 2023-07-19 at 8 13 17 AM

Before|After
---|---
Screenshot 2023-07-19 at 8 16 31 AM|Screenshot 2023-07-19 at 8 13 28 AM

Before|After
---|---
Screenshot 2023-07-19 at 8 16 44 AM|Screenshot 2023-07-19 at 8 13 39 AM

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [x] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [x] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [x] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Review by lougoncharenko — APPROVED** [review]
I applied a couple different font families in the theme settings and experimented with filling pout different input fields for testing. With each font-family change, the styling of the text in the field inputs also changed. This change looks visually appealing and matches the rest of the styling of the merchant's page

File: assets/base.css [hunk 1]
```diff
@@ -1585,6 +1585,7 @@ details[open] > .share-button__fallback {
   appearance: none;
   background-color: rgb(var(--color-background));
   color: rgb(var(--color-foreground));
+  font-family: var(--font-body-family);
```
[the remark this answers is not in the document]
Comment (kjellr) [reply assets/base.css:1588]: Ah, great catch!

Comment (ludoboludo) [inline assets/base.css:1588]: I think here what Kai mentioned is still missing. For example we can see the different on the contact page/form: 
<details>

<summary>Screenshot</summary>



</details>

**Review by ludoboludo — COMMENTED** [review]
Don't forget to change the branch target. Right now it's still pointing to a release branch instead of main 👍 

I just have one comment to address Kai's point otherwise looks god.

**Pull request merged**